src/store/users/reducer.js

```diff
@@ -0,0 +1,25 @@
+import * as types from 'kitsu/store/types';
+
+const initialState = {};
+
+export const usersReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case types.CAPTURE_USERS_DATA: {
+      const users = {};
+      action.payload.filter(u => u.type !== 'user').map(u => (users[u.id] = u));
+      return {
+        ...state,
+        ...users,
+      };
+    }
+    case types.USER_FOLLOW_SUCCESS: {
+      const { id } = action.payload;
+      state[id] = { ...state[id], following: true };
```

Comment:
If we're following [immutable update patterns](http://redux.js.org/docs/recipes/reducers/ImmutableUpdatePatterns.html#immutable-update-patterns), this should be changed.

Reply:
@edencakir the state is getting mutated in the local scope, I'm returning a new copy of the `state` (check return code) which is essentially what redux expects.